### PR TITLE
added scripts/wrappers to simplify the process of running adv_im

### PR DESCRIPTION
**Pull request by shot0829**
#### Checklist
- Have you updated the README (yes/no)?
- Have you updated the CHANGELOG (yes/no)?

### CODE REVIEW

File: scripts/run_adv_im_obs_maui.sl
Comment (jpa198) [inline scripts/run_adv_im_obs_maui.sl:14]: Should be in the scripts/nesi_scripts directory

File: scripts/run_adv_im_obs_maui.sl
Comment (jpa198) [inline scripts/run_adv_im_obs_maui.sl:76]: Can't run a completion check after it's done?
Even just counting the right number of stations are in each csv?

File: scripts/run_adv_im.sh
Comment (jpa198) [inline scripts/run_adv_im.sh:18]: These should be moved to /nesi/project, rather than left in nobackup?

Comment (shot0829) [reply scripts/run_adv_im.sh:18]: i'll run a du to check the total size on BBs and Models. If its too large we might have to keep it there.

**Review by jasonmotha — COMMENTED** [review]
I don't like having an a-version of gmsim, but until we've got this tested doing a full run it's the better option.

**Review by jasonmotha — CHANGES_REQUESTED** [review]

File: scripts/nesi_scripts/run_adv_im.sh
Comment (jasonmotha) [inline scripts/nesi_scripts/run_adv_im.sh:199]: function rather than repeat code.

**Review by jasonmotha — CHANGES_REQUESTED** [review]

File: shared_workflow/install_shared.py
Comment (jasonmotha) [inline shared_workflow/install_shared.py:97]: These aren't actually saved to the vm config in srfinfo2vm. Are they written elsewhere before install?

**Review by jasonmotha — CHANGES_REQUESTED** [review]

File: shared_workflow/install_shared.py
Comment (shot0829) [reply shared_workflow/install_shared.py:97]: its generated by gen_coords

Comment (shot0829) [reply shared_workflow/install_shared.py:97]: i'm moving them to gen_coords, or at least somewhere in Pre-processing. ( since its gen_coords that generates them)
the current workflow doesnt make sense and cause unexpected file corruptions.
what we have now in install_cybershake_fault:
1. validate_vm
2.install sim folder
2a. hard-code and  then change/dump yaml config file (invalids the validation just happened)

IMO, the install process should not be touching files in Data/Model folder
What raised this issue is that there was a crash during install phase, and it corrupted/nuked the yaml file with bad info

Comment (jasonmotha) [reply shared_workflow/install_shared.py:97]: Sweet. I'll resolve this conversation once the pull-request in pre-pro is merged.
Is gen-coords going to be called from srfinfo2vm? Or what are you thinking?

Comment (shot0829) [reply shared_workflow/install_shared.py:97]: gen_coords is already called by srfinfo2vm, but currently only generates the files (e.g gridfile, gridout etc) without updating the vm_params.yaml.
I'm thinking adding in the update function in gen_coords.

Comment (shot0829) [reply shared_workflow/install_shared.py:97]: https://github.com/ucgmsim/Pre-processing/pull/173

File: scripts/run_adv_im.sh
Comment (shot0829) [reply scripts/run_adv_im.sh]: https://github.com/ucgmsim/slurm_gm_workflow/pull/337/commits/d1f96e122d5eb707a01e9182039847c971c9d9a7

File: scripts/run_adv_im_obs_maui.sl
Comment (shot0829) [reply scripts/run_adv_im_obs_maui.sl]: updated to include the comprehensive test
https://github.com/ucgmsim/slurm_gm_workflow/pull/337/commits/1650dfed18c3a9b75e4ca77f49ac7890c331af62

File: scripts/nesi_scripts/run_adv_im.sh
Comment (shot0829) [reply scripts/nesi_scripts/run_adv_im.sh]: https://github.com/ucgmsim/slurm_gm_workflow/pull/337/commits/3cf0f2575dc469150c0333b1b5d1b878eecbe9f6

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (jpa198) [inline scripts/nesi_scripts/adv_im/run_adv_im.sh:126]: Add a comment or a print to say why we crashed
Comment so that someone reading this script can understand the possible failures at this point

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (jpa198) [inline scripts/nesi_scripts/adv_im/run_adv_im.sh:145]: Same as above, comment or print to say why this crashed

Comment (jpa198) [reply scripts/nesi_scripts/adv_im/run_adv_im.sh:145]: Also exit code?

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (jpa198) [inline scripts/nesi_scripts/adv_im/run_adv_im.sh:190]: Max retry counter? Otherwise this could go forever

File: scripts/nesi_scripts/adv_im/run_adv_im_obs_maui.sl
Comment (jpa198) [inline scripts/nesi_scripts/adv_im/run_adv_im_obs_maui.sl:27]: Default value?

File: scripts/nesi_scripts/run_adv_im.sh
Comment (shot0829) [reply scripts/nesi_scripts/run_adv_im.sh]: https://github.com/ucgmsim/slurm_gm_workflow/pull/337/commits/c3f2c49681349dba424c941fcf6ca4cca4655850

File: scripts/clean_restart.sh
Comment (shot0829) [reply scripts/clean_restart.sh]: cannot find which comment solved this, but comments are added in the newer commits
https://github.com/ucgmsim/slurm_gm_workflow/blob/d1f96e122d5eb707a01e9182039847c971c9d9a7/scripts/nesi_scripts/run_adv_im.sh

**Review by jpa198 — CHANGES_REQUESTED** [review]

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (jasonmotha) [reply scripts/nesi_scripts/adv_im/run_adv_im.sh:190]: attempt

**Review by jasonmotha — CHANGES_REQUESTED** [review]

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (shot0829) [reply scripts/nesi_scripts/adv_im/run_adv_im.sh:190]: my original thought was that the user should decide when they are going to kill it or not.
this error (failed to start a screen) rarely happens, and my guess was that its related to if anyone is stressing the login node, thus usually naturally resolve itself after a few try. ( hence the unconditioned loop until success)

File: scripts/nesi_scripts/adv_im/run_adv_im_obs_maui.sl
Comment (shot0829) [reply scripts/nesi_scripts/adv_im/run_adv_im_obs_maui.sl:27]: yeah, the :- basically means if none then use this.

File: shared_workflow/install_shared.py
Comment (shot0829) [reply shared_workflow/install_shared.py:100]: yeah, some of the changes should be break into a different PR, instead of being in this huge one.
i'm cleaning it up and will move less related ones into separate PR

File: scripts/remove_restart_from_tar.sh
Comment (shot0829) [reply scripts/remove_restart_from_tar.sh:2]: the newest clean-up script should be handling this properly i.e. removing the restart before taring.
this script is more for cleaning/reducing the size of old tar files.
as mentioned above, i'll be moving less related scripts into a new PR

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (shot0829) [reply scripts/nesi_scripts/adv_im/run_adv_im.sh:114]: the workflow is bacially copying/linking the previous ran BB.bin into a new location, so it will not affect anything in previous runs. and in order to make our automated script submit the jobs, we need to make sure the mgmt_db have all the steps and in clean-state . And i think creating a new one is the easiest and safest way to do so.

File: templates/gmsim/18.5.3.1.a/root_defaults.yaml
Comment (jpa198) [reply templates/gmsim/18.5.3.1.a/root_defaults.yaml:1]: If the only difference with 18.5.3.1 is the adv_IM section, then no it'll be ignored by everything in workflow. This can go into 18.5.3.1

File: templates/gmsim/18.5.3.1.a/root_defaults.yaml_template
Comment (jpa198) [inline templates/gmsim/18.5.3.1.a/root_defaults.yaml_template:20]: Why does this file exist?

**Comment by shot0829** [comment]
chaning to draft, as there are many scripts/file that are either redundant or needs to be move into another PR.

File: scripts/link_bb.sh
Comment (shot0829) [reply scripts/link_bb.sh:1]: this is a generic code that can be used across all platform. and can be used outside of adv_im. 
so i dont really know which sub-folder it should belong. and why do you think it should be in a separate folder?

**Comment by shot0829** [comment]
closing PR, re-opening it once its in a more completed stat

**Comment by shot0829** [comment]
re-opening PR

File: scripts/nesi_scripts/adv_im/clean_adv_im_tmps.sh
Comment (jpa198) [inline scripts/nesi_scripts/adv_im/clean_adv_im_tmps.sh:1]: Is this script nesi specific? If not there should be a bash_scripts folder to put the agnostic non-python scripts in

File: scripts/nesi_scripts/adv_im/update_root_yaml.py
Comment (jasonmotha) [inline scripts/nesi_scripts/adv_im/update_root_yaml.py:18]: path_root_yaml

File: scripts/nesi_scripts/adv_im/update_root_yaml.py
Comment (jasonmotha) [inline scripts/nesi_scripts/adv_im/update_root_yaml.py:31]: You'll need to black this.

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (jasonmotha) [inline scripts/nesi_scripts/adv_im/run_adv_im.sh:257]: Does split list really need to be it's own file?

File: scripts/nesi_scripts/adv_im/run_adv_im.sh
Comment (jasonmotha) [inline scripts/nesi_scripts/adv_im/run_adv_im.sh:252]: I think it's probably worth having a separate workflow for observed rather than including them in simulation. Because we should be only running observed once rather than for multiple simulations.

Comment (jasonmotha) [reply scripts/nesi_scripts/adv_im/run_adv_im.sh:252]: Thoughts @jpa198

**Review by jasonmotha — CHANGES_REQUESTED** [review]

File: scripts/nesi_scripts/adv_im/clean_adv_im_tmps.sh
Comment (jasonmotha) [inline scripts/nesi_scripts/adv_im/clean_adv_im_tmps.sh:2]: I think that this should be part of "clean-up" step rather than as an individual script. (or as a separate adv-im cleanup step)

**Pull request closed without merging**
